src/mlpack/core/tree/rectangle_tree/rectangle_tree.hpp

```diff
@@ -424,7 +419,10 @@ class RectangleTree
    *
    * @param index Index of point for which a dataset index is wanted.
    */
-  size_t Point(const size_t index) const;
+  const size_t& Point(const size_t index) const { return points[index]; }
```

Comment:
I don't think there's any reason to return a reference here, since that is the same size as a size_t, so I'll go ahead and change this back after merge unless there is a reason I shouldn't that I am not aware of.